src/java/com/rapleaf/jack/transaction/DbPoolManager.java

```diff
@@ -0,0 +1,153 @@
+package com.rapleaf.jack.transaction;
+
+import java.util.NoSuchElementException;
+import java.util.concurrent.Callable;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.PooledObjectFactory;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rapleaf.jack.IDb;
+import com.rapleaf.jack.exception.ConnectionCreationFailureException;
+import com.rapleaf.jack.exception.NoAvailableConnectionException;
+
+class DbPoolManager<DB extends IDb> implements IDbManager<DB> {
+  private static final Logger LOG = LoggerFactory.getLogger(DbPoolManager.class);
+
+  public static int DEFAULT_MAX_TOTAL_CONNECTIONS = 1;
```

Comment:
I will change it to `3` then.

Bruce, if this is a required parameter, to enable unlimited number of connections, people have to specify a negative value, which is confusing and I'd like to avoid that. The builder has an `enableInfiniteConnections` method to do that.